tests/test_bucket_map.py

```diff
@@ -307,3 +463,601 @@ def test_get_required_groups(sample_bucket_map):
+                "Condition": {
+                    "StringNotLike": {
+                        "s3:prefix": [
+                            "science/",
+                            "science2/",
+                            "science3/",
+                            "science4/"
+                        ]
+                    }
+                }
+            },
+            # This statement is redundant, but makes the implementation simpler
+            {
+                "Effect": "Allow",
+                "Action": ["s3:GetObject", "s3:ListBucket"],
+                "Resource": [
+                    "arn:aws:s3:::bucket-name",
+                    "arn:aws:s3:::bucket-name/*"
+                ],
+                "Condition": {
+                    "StringLike": {
+                        "s3:prefix": [
+                            "browse/"
+                        ]
+                    }
+                }
+            },
+        ]
+    }
+
+
+def test_to_iam_policy(sample_bucket_map):
+    b_map = BucketMap(sample_bucket_map, bucket_name_prefix="pre-")
+
+    assert b_map.to_iam_policy(groups=()) == {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": ["s3:GetObject", "s3:ListBucket"],
+                "Resource": [
+                    "arn:aws:s3:::pre-authed-bucket",
+                    "arn:aws:s3:::pre-authed-bucket/*",
+                    "arn:aws:s3:::pre-browse-bucket",
+                    "arn:aws:s3:::pre-browse-bucket/*",
+                    "arn:aws:s3:::pre-nested-bucket-public",
+                    "arn:aws:s3:::pre-nested-bucket-public/*",
+                    "arn:aws:s3:::pre-nested-bucket-private",
+                    "arn:aws:s3:::pre-nested-bucket-private/*",
+                ]
+            },
+            {
+                "Effect": "Allow",
+                "Action": ["s3:GetObject", "s3:ListBucket"],
+                "Resource": [
+                    "arn:aws:s3:::pre-bucket",
+                    "arn:aws:s3:::pre-bucket/*"
+                ],
+                "Condition": {
+                    "StringNotLike": {
+                        "s3:prefix": ["2020/12"]
+                    }
+                }
+            },
+            # This statement is redundant, but makes the implementation simpler
+            {
+                "Effect": "Allow",
+                "Action": ["s3:GetObject", "s3:ListBucket"],
+                "Resource": [
+                    "arn:aws:s3:::pre-bucket",
+                    "arn:aws:s3:::pre-bucket/*"
+                ],
+                "Condition": {
+                    "StringLike": {
+                        "s3:prefix": ["browse"]
+                    }
+                }
+            }
+        ]
+    }
+
+
+@pytest.mark.parametrize("groups", (None, (), ("foo")))
+def test_to_iam_policy_groups_noaccess(groups_bucket_map, groups):
+    b_map = BucketMap(groups_bucket_map)
```

Comment:
Any reason to not generate the BucketMap obj as a fixture instead of at the start of each test?

Reply:
Well for these it's because they're testing that the same bucket map will generate different policies when you have different EDL groups. So I feel like it makes sense to use a fixture so that each test is guaranteed to have the same input data.

Reply:
right, I am asking why not
```python
@pytest.fixture
def groups_bucket_map():
   map =  {
        "PATH1": "bucket1",
        "PATH2": "bucket2",
        "PATH3": "bucket3",
        "PRIVATE_BUCKETS": {
            "bucket1": ["group1"],
            "bucket2": ["group2"],
            "bucket3": ["group3"],
        }
    }
    return BucketMap(map)
```

Reply:
In the places you use it, you just repeat the same ```b_map = BucketMap(groups_bucket_map)```

Reply:
Oh I see. I guess I just did it this way for consistency with the other tests. Makes copy/pasting easier :P